src/LoginRouter.js

```diff
@@ -118,7 +118,8 @@ function (BaseLoginRouter,
       'signin/verify/u2f': 'verifyU2F',
       'signin/verify/generic_saml/assertion:saml2': 'verifySAMLFactor',
       'signin/verify/generic_oidc/assertion:oidc': 'verifyOIDCFactor',
-      'signin/verify/:provider/:factorType': 'verify',
+      'signin/verify/:factorType': 'verifyNoProvider',
+      'signin/verify/:provider/:factorType(/:factorIndex)': 'verify',
```

Comment:
it is an optional part of the url. Basically when we have 2 or more push factors, we show all of them in the dropdown. The way the widget works is that we have one route per factor (e.g. `verify/okta/push`, `verify/okta/sms`), so if we have multiple push factors one single route per factorType won't work.
There were 2 options to solve this:
- add something extra to the route, so we can differentiate each push factor. This is what I implemented. Adding an index (out of all the push factors, which one is the one that is selected) -> `verify/okta/push/1` 
- have the appState know which factor is selected. I didn't do this because this means that we need extra logic in a few places to know that selecting a factor in the dropdown does not navigate to a new route, but instead it should just "refresh" (re-render) the current controller. Also it will need to mimic a factor transition, and also that means it won't have an entry in the history... this was a little more complicated in my opinion.